File: backend/app/Models/Seller.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Seller extends Model
{
    protected $fillable = [
        'user_id',
        'business_name',
        'line_of_business',
        'id_image',
        'business_permit',
        'approval_status',
        'status',
    ];

    protected $casts = [
        'user_id' => 'integer',
    ];

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function products()
    {
        return $this->hasMany(Product::class, 'seller_id');
    }

    public function isApproved(): bool
    {
        return $this->approval_status === 'approved' && $this->status === 'active';
    }
}

File: frontend/lib/models/seller.dart
class Seller {
  final int id;
  final int userId;
  final String businessName;
  final String lineOfBusiness;
  final String? idImage;
  final String? businessPermit;
  final String approvalStatus;
  final String status;

  Seller({
    required this.id,
    required this.userId,
    required this.businessName,
    required this.lineOfBusiness,
    this.idImage,
    this.businessPermit,
    required this.approvalStatus,
    required this.status,
  });

  bool get isApproved => approvalStatus == 'approved' && status == 'active';
  bool get isPending => approvalStatus == 'pending';

  factory Seller.fromJson(Map<String, dynamic> json) => Seller(
        id: json['id'] as int,
        userId: json['user_id'] as int? ?? 0,
        businessName: json['business_name'] as String? ?? '',
        lineOfBusiness: json['line_of_business'] as String? ?? '',
        idImage: json['id_image'] as String?,
        businessPermit: json['business_permit'] as String?,
        approvalStatus: json['approval_status'] as String? ?? 'pending',
        status: json['status'] as String? ?? 'active',
      );
}